lib/transport/control.ts

```diff
@@ -66,17 +66,25 @@ export interface Subscribe {
 	namespace: string
 	name: string
 
-	start_group: Location
-	start_object: Location
-	end_group: Location
-	end_object: Location
+	filter_type: FilterType
+
+	// start_group?: Location
+	// start_object?: Location
+	// end_group?: Location
+	// end_object?: Location
 
 	params?: Parameters
 }
 
-export interface Location {
-	mode: "none" | "absolute" | "latest" | "future"
-	value?: number // ignored for type=none, otherwise defaults to 0
+export interface FilterType {
+	// TODO: discuss un-nesting start/end values
```

Comment:
I spent some time writing this out as un-nested, but it required more code and didn't offer any obvious benefits. I think I probably end up going full circle here and just calling this whole field "Location" again.